remote-api/src/test/java/org/alfresco/rest/api/model/rules/RuleTest.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * #%L
+ * Alfresco Remote API
+ * %%
+ * Copyright (C) 2005 - 2022 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+
+package org.alfresco.rest.api.model.rules;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+
+import org.alfresco.repo.action.ActionConditionImpl;
+import org.alfresco.repo.action.ActionImpl;
+import org.alfresco.repo.action.executer.ScriptActionExecuter;
+import org.alfresco.service.Experimental;
+import org.alfresco.service.cmr.action.Action;
+import org.alfresco.service.cmr.action.ActionCondition;
+import org.alfresco.service.cmr.repository.NodeRef;
+import org.alfresco.service.cmr.repository.StoreRef;
+import org.alfresco.service.cmr.rule.RuleType;
+import org.assertj.core.api.Condition;
+import org.junit.Test;
+
+@Experimental
+public class RuleTest
+{
+    private static final String RULE_ID = "fake-rule-id";
+    private static final String RULE_NAME = "rule name";
+    private static final String RULE_DESCRIPTION = "rule description";
+    private static final boolean RULE_ENABLED = true;
+    private static final boolean RULE_CASCADE = true;
+    private static final boolean RULE_ASYNC = false;
+    private static final boolean RULE_SHARED = true;
+
+    @Test
+    public void testFrom()
+    {
+        final org.alfresco.service.cmr.rule.Rule ruleModel = createRuleModel();
+
+        // when
+        final Rule rule = Rule.from(ruleModel, RULE_SHARED);
+
+        assertThat(rule).is(havingExpectedConstantValues());
```

Comment:
I'm not sure this is easier to read with the assertions in the private method.  I don't think you're reusing the method either.  Does this maybe help by giving more information in the case of multiple errors?

Reply:
Fixed that by comparing actual rule with predefined expected rule.